tests/compiler/retain-i32.optimized.wat

```diff
@@ -1,18 +1,241 @@
 (module
  (type $none_=>_none (func))
+ (type $i32_i32_=>_none (func (param i32 i32)))
+ (type $i32_i32_i32_i32_=>_none (func (param i32 i32 i32 i32)))
+ (import "env" "abort" (func $~lib/builtins/abort (param i32 i32 i32 i32)))
  (memory $0 1)
  (data (i32.const 1024) "\1a\00\00\00\01\00\00\00\01\00\00\00\1a\00\00\00r\00e\00t\00a\00i\00n\00-\00i\003\002\00.\00t\00s")
  (export "memory" (memory $0))
  (start $~start)
- (func $~start
+ (func $retain-i32/test (param $0 i32) (param $1 i32)
+  (local $2 i32)
+  local.get $0
+  local.get $1
+  i32.add
+  local.tee $2
+  i32.extend8_s
+  local.get $2
+  i32.extend8_s
+  i32.ne
```

Comment:
this definitely isolate expressions from pattern matching for `x != x` => `false`. Btw great challenge for improve binaryen in this part.